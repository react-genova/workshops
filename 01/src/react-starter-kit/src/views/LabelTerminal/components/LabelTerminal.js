//@flow
import React from 'react';
import styled from 'styled-components';

type Props = {
    text:string
};

const Container = styled.div`
    display: flex;
    flex-direction: row;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.2em;
    background-color: #202020;
    padding: 0.06em;
`;

const Prompt = styled.div`
    color: white;
    padding-left: 0.2em;
    padding-right: 0.2em;
`;

const Command = styled.div`
    color: #A0A0A0;
`;

const LabelTerminal = ({text}:Props) => (
    <Container>
        <Prompt>{">"}</Prompt>
        <Command>{text}</Command>
    </Container>
);

export default LabelTerminal;