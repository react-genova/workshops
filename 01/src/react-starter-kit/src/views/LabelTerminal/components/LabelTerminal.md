### Single Label Terminal ###

```js
<LabelTerminal text="npm install --save redux" />
```

### Multiple Label Terminal ###

```js
<div>
<LabelTerminal text="npm install --save redux" />
<LabelTerminal text="npm install --save redux-effects" />
</div>
```