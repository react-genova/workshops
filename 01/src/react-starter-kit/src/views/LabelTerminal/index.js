//@flow
export {default} from './components/LabelTerminal';