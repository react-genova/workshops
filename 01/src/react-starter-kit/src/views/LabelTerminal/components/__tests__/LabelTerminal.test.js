//@flow
import React from 'react';
import LabelTerminal from '../LabelTerminal';
import { shallow } from 'enzyme';

describe('LabelTerminal component', () => {

    it('simple label snapshot', () => {
        const component = shallow(<LabelTerminal text="npm install --save redux" />);
        expect(component).toMatchSnapshot();
    });

});