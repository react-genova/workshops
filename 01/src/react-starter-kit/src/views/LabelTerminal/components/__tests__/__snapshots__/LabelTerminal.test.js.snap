// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelTerminal component simple label snapshot 1`] = `
<styled.div>
  <styled.div>
    &gt;
  </styled.div>
  <styled.div>
    npm install --save redux
  </styled.div>
</styled.div>
`;
